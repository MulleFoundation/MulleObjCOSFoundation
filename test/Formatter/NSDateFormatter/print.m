//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjCOSFoundation/MulleObjCOSFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif



int   main( int argc, const char * argv[])
{
   NSDateFormatter   *formatter;
   NSCalendarDate    *date;
   NSTimeZone        *timeZone;

#ifdef __MULLE_OBJC__
   if( mulle_objc_global_check_universe( __MULLE_OBJC_UNIVERSENAME__) != mulle_objc_universe_is_ok)
      return( 1);
#endif

   formatter = [[[NSDateFormatter alloc] initWithDateFormat:@"%Y-%m-%dT%H:%M:%SZ%z"
                                       allowNaturalLanguage:NO] autorelease];

   timeZone = [NSTimeZone timeZoneForSecondsFromGMT:3*60*60];
#ifdef __MULLE_OBJC__
   date     = [[[NSCalendarDate alloc] mulleInitWithTimeIntervalSinceReferenceDate:(24*60*60)*33+(01*60*60)+(5*60)+6
                                                                          timeZone:timeZone] autorelease];
#else
   date     = [[[NSCalendarDate alloc] initWithTimeIntervalSinceReferenceDate:(24*60*60)*33+(01*60*60)+(5*60)+6] autorelease];
   [date setTimeZone:timeZone];
#endif
   printf( "%s (%.f)\n", [[formatter stringFromDate:date] UTF8String], [date timeIntervalSinceReferenceDate]);

   return( 0);
}
