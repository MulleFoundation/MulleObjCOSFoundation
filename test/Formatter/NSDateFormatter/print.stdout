2001-02-03T04:05:06Z+0300 (2855106)
